### PR TITLE
Fix specs to support templating feature

### PR DESCRIPTION
**Pull request by ivanilves**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -116,9 +116,10 @@ docker-image: build
 docker-image:
 	@docker image build --no-cache -t ${DOCKER_REPO}:${DOCKER_TAG} .
 
+build: LDFLAGS=$(shell test `uname -s` = "Darwin" && echo '-s -w' || echo '-d -s -w')
```
Comment (ivanilves) [inline Makefile]: This is to make build work on my new Mac 🙂

File: api/v1/v1_test.go [hunk 2]
```diff
@@ -43,7 +43,12 @@ func runEnd2EndJob(pullRefs, seedRefs []string) ([]string, error) {
 		}
 	}
 
-	pushConfig := PushConfig{Registry: registryContainer.Hostname(), PathSeparator: "/", PathTemplate: "{{ .Prefix }}{{ .Path }}"}
+	pushConfig := PushConfig{
+		Registry:      registryContainer.Hostname(),
+		PathSeparator: "/",
+		PathTemplate:  "{{ .Prefix }}{{ .Path }}",
+		TagTemplate:   "{{ .Tag }}",
```
Comment (ivanilves) [inline api/v1/v1_test.go]: We need to add `TagTemplate:   "{{ .Tag }}",` in specs.

File: api/v1/v1_test.go [hunk 3]
```diff
@@ -120,9 +125,9 @@ func TestEnd2End(t *testing.T) {
 		pushRefs, err := runEnd2EndJob(testCase.pullRefs, testCase.seedRefs)
 
 		if testCase.isCorrect {
-			assert.Nil(err, "should be no error")
+			assert.Nil(err, fmt.Sprintf("should be no error (%+v)", testCase))
 		} else {
-			assert.NotNil(err, "should be an error")
+			assert.NotNil(err, fmt.Sprintf("should be an error (%+v)", testCase))
```
Comment (ivanilves) [inline api/v1/v1_test.go]: This is to have more observability on what happened during failed tests.

Comment (zealic) [reply api/v1/v1_test.go]: Very nice!!!

File: api/v1/v1_test.go (hunk 2 above)
Comment (vonrabbe) [reply api/v1/v1_test.go]: 👍

**Pull request merged**